packages/terra-form-input/tests/wdio/form-input-spec.js

```diff
@@ -0,0 +1,132 @@
+/* global browser, Terra */
+describe('Form-Input', () => {
+  describe('Valid', () => {
+    beforeEach(() => {
+      browser.url('/#tests/form-input-tests/default');
+    });
+
+    Terra.should.beAccessible();
+    Terra.should.matchScreenshot();
+    Terra.should.themeEachCustomProperty({
+      '--terra-form-input-background-color': 'red',
+      '--terra-form-input-focus-animation-background': 'linear-gradient(rgba(0, 0, 255, 0.5), rgba(255, 255, 0, 0.5)',
+      '--terra-form-input-border-radius': '0.3rem',
+      '--terra-form-input-border': '2px dashed #4a4a41',
+      '--terra-form-input-box-shadow': 'inset 5em 1em gold',
+      '--terra-form-input-color': '#fff',
+      '--terra-form-input-font-size': '2.143rem',
+      '--terra-form-input-line-height': '2.143',
+      '--terra-form-input-opacity': '0.5',
+      '--terra-form-input-padding-bottom': '1.571rem',
+      '--terra-form-input-padding-left': '1.357rem',
+      '--terra-form-input-padding-right': '1.357rem',
+      '--terra-form-input-padding-top': '1.571rem',
+      '--terra-form-input-transition-duration': '2s',
+      '--terra-form-input-transition-timing-function': 'ease-out',
+      '--terra-form-input-focus-background-start': '20px 20px',
+    });
+  });
+
+  describe('Invalid', () => {
+    beforeEach(() => {
+      browser.url('/#tests/form-input-tests/invalid');
+    });
+
+    Terra.should.beAccessible();
+    Terra.should.matchScreenshot();
+    Terra.should.themeEachCustomProperty({
+      '--terra-form-input-error-focus-animation-background': 'linear-gradient(rgba(0, 0, 255, 0.5), rgba(255, 255, 0, 0.5)',
+      '--terra-form-input-error-border-radius': '20rem',
+      '--terra-form-input-error-border': '20rem dashed blue',
+      '--terra-form-input-error-opacity': '0.5',
+    });
+  });
+
+  describe('Enabled', () => {
+    describe('Enabled Non-Selected', () => {
+      beforeEach(() => {
+        browser.url('/#tests/form-input-tests/default');
+      });
+
+      Terra.should.beAccessible();
+      Terra.should.matchScreenshot();
+    });
+
+    describe('Enabled Hovered', () => {
+      beforeEach(() => {
+        browser.url('/#tests/form-input-tests/default');
+        browser.moveToObject('input');
+      });
+
+      Terra.should.beAccessible();
+      Terra.should.matchScreenshot();
+      Terra.should.themeEachCustomProperty({
+        '--terra-form-input-hover-border': '20rem dashed blue',
+      });
+    });
+
+    describe('Enabled Keyboard-Focused', () => {
+      beforeEach(() => {
+        browser.url('/#tests/form-input-tests/default');
+        browser.keys('Tab');
```

Comment:
Rather than tab to focus on the input, we should emulate a focus state by creating an `is-focused` class that applies the same styles as `:focus`. Then we need to create a test page for focus and set the `is-focused` class on the input. This is to get around the issue @saedar noticed where these tests have a potential to fail based on if the caret to indicate where the cursor is, is visible or not as it blinks.

This would allow us to capture focus styles without the blinking caret in the input.

Reply:
Done 8918a8ddbdc438465dd5bab14b39ee7723b03652